### PR TITLE
d/aws_route: 'destination_prefix_list_id' attribute set for managed prefix list destinations

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Builds on: #14014, #16930, #16961, #17291
Dependents: None (yet)

Relates: #14018, #15273.

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSRouteDataSource_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteDataSource_ -timeout 180m
=== RUN   TestAccAWSRouteDataSource_basic
=== PAUSE TestAccAWSRouteDataSource_basic
=== RUN   TestAccAWSRouteDataSource_TransitGatewayID
=== PAUSE TestAccAWSRouteDataSource_TransitGatewayID
=== RUN   TestAccAWSRouteDataSource_IPv6DestinationCidr
=== PAUSE TestAccAWSRouteDataSource_IPv6DestinationCidr
=== RUN   TestAccAWSRouteDataSource_LocalGatewayID
=== PAUSE TestAccAWSRouteDataSource_LocalGatewayID
=== RUN   TestAccAWSRouteDataSource_CarrierGatewayID
=== PAUSE TestAccAWSRouteDataSource_CarrierGatewayID
=== RUN   TestAccAWSRouteDataSource_DestinationPrefixListId
=== PAUSE TestAccAWSRouteDataSource_DestinationPrefixListId
=== RUN   TestAccAWSRouteDataSource_GatewayVpcEndpoint
=== PAUSE TestAccAWSRouteDataSource_GatewayVpcEndpoint
=== CONT  TestAccAWSRouteDataSource_basic
=== CONT  TestAccAWSRouteDataSource_CarrierGatewayID
--- PASS: TestAccAWSRouteDataSource_CarrierGatewayID (25.83s)
=== CONT  TestAccAWSRouteDataSource_GatewayVpcEndpoint
=== CONT  TestAccAWSRouteDataSource_DestinationPrefixListId
--- PASS: TestAccAWSRouteDataSource_GatewayVpcEndpoint (41.91s)
--- PASS: TestAccAWSRouteDataSource_DestinationPrefixListId (199.51s)
=== CONT  TestAccAWSRouteDataSource_IPv6DestinationCidr
--- PASS: TestAccAWSRouteDataSource_IPv6DestinationCidr (24.39s)
=== CONT  TestAccAWSRouteDataSource_LocalGatewayID
    data_source_aws_outposts_outposts_test.go:67: skipping since no Outposts found
--- SKIP: TestAccAWSRouteDataSource_LocalGatewayID (0.64s)
=== CONT  TestAccAWSRouteDataSource_TransitGatewayID
--- PASS: TestAccAWSRouteDataSource_basic (305.32s)
--- PASS: TestAccAWSRouteDataSource_TransitGatewayID (378.62s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	671.027s
```

### CODE REVIEW

File: aws/data_source_aws_route_test.go [hunk 1]
```diff
@@ -73,69 +112,108 @@ func TestAccAWSRouteDataSource_LocalGatewayID(t *testing.T) {
...
+}
 
-		if attr["route_table_id"] != rts.Primary.Attributes["id"] {
-			return fmt.Errorf(
-				"route_table_id is %s; want %s",
-				attr["route_table_id"],
-				rts.Primary.Attributes["id"],
-			)
-		}
+func TestAccAWSRouteDataSource_GatewayVpcEndpoint(t *testing.T) {
```
Comment (ewbankkit) [inline aws/data_source_aws_route_test.go:159]: This test validates [this change](https://github.com/hashicorp/terraform-provider-aws/pull/17295/files#diff-b8af03892138451662b5730171b62cc571d6964da7026db591b276a37c5dc81fR123-R127) by adding a Gateway VPC Endpoint (S3), specifying that endpoint's prefix list ID to the data source and verifying that no route is returned.

**Comment by ewbankkit** [comment]
##### TODO

Add acceptance test pre-checks: #16806.

**Comment by ewbankkit** [comment]
The `XAT001: missing ErrorCheck` linter errors should be resolved once #17291 is merged and this PR is rebased.

**Review by YakDriver — APPROVED** [review]
Looks great! 🎉 

Acceptance tests in commercial:
```
--- SKIP: TestAccAWSRouteDataSource_LocalGatewayID (1.39s)
--- PASS: TestAccAWSRouteDataSource_CarrierGatewayID (25.22s)
--- PASS: TestAccAWSRouteDataSource_IPv6DestinationCidr (25.27s)
--- PASS: TestAccAWSRouteDataSource_GatewayVpcEndpoint (41.80s)
--- PASS: TestAccAWSRouteDataSource_basic (89.40s)
--- PASS: TestAccAWSRouteDataSource_DestinationPrefixListId (219.00s)
--- PASS: TestAccAWSRouteDataSource_TransitGatewayID (380.64s)
```

Acceptance tests in GovCloud:
```
--- SKIP: TestAccAWSRouteDataSource_DestinationPrefixListId (1.59s)
--- SKIP: TestAccAWSRouteDataSource_CarrierGatewayID (1.61s)
--- SKIP: TestAccAWSRouteDataSource_LocalGatewayID (1.86s)
--- PASS: TestAccAWSRouteDataSource_IPv6DestinationCidr (27.01s)
--- PASS: TestAccAWSRouteDataSource_GatewayVpcEndpoint (44.53s)
--- PASS: TestAccAWSRouteDataSource_basic (111.36s)
--- PASS: TestAccAWSRouteDataSource_TransitGatewayID (351.68s)
```

**Pull request merged**